src/app.html

```diff
@@ -0,0 +1,101 @@
+                                <i class="material-icons">search</i>
+                            </label>
+                            <input value={{searchText}} id="PVLSearchInput" type="text" class="form-control border-left-0" placeholder="Search..." aria-label="Search" aria-describedby="basic-addon1">
+                        </div>
+                    </div>
+                </form>
+            </div>
+        </div>
+
+        <div class="row mt-1 | product-list">
+            <div class="col">
+                <table class="table {{sortDesc ? 'sort-desc' : ''}}">
+                    <thead>
+                        <tr>
+                            <th scope="col">
+                                <label class="custom-control custom-checkbox">
+                                    <input type="checkbox" class="custom-control-input" checked={{editingAll}}>
+                                    <span class="custom-control-indicator"></span>
+                                </label>
+                            </th>
+                            <th scope="col" colspan="2">
+                                Name
+                                <i class="material-icons {{sortColumn !== 'name' ? 'text-muted' : 'sort'}}" on-click=@this.sortBy('name')>arrow_drop_down</i>
+                            </th>
+                            <th scope="col" class="type-column-header text-center">
+                                Type
+                                <i class="material-icons {{sortColumn !== 'type' ? 'text-muted' : 'sort'}}" on-click=@this.sortBy('type')>arrow_drop_down</i>
+                            </th>
+                            <th scope="col" class="price-column-header text-center">
+                                Price
+                                <i class="material-icons {{sortColumn !== 'price' ? 'text-muted' : 'sort'}}" on-click=@this.sortBy('price')>arrow_drop_down</i>
+                            </th>
+                            <th scope="col" class="inventory-column-header text-center">
+                                Inventory
+                                <i class="material-icons {{sortColumn !== 'inventory' ? 'text-muted' : 'sort'}}" on-click=@this.sortBy('inventory')>arrow_drop_down</i>
+                            </th>
+                        </tr>
+                    </thead>
+                    <tbody class="text-muted">
+                        {{#each visibleProducts}}
+                            {{#if editing[.id]}}
+                                {{#editing[.id]}}
+                                    {{>productRowEdit}}
+                                {{/editing[.id]}}
+                            {{else}}
+                                {{>productRow}}
+                            {{/if}}
+                        {{/each}}
+                        {{#each newProducts}}
+                            {{>productRowEdit}}
+                        {{/each}}
```

Comment:
In this section, we utilize [html partials](https://ractive.js.org/api/#in-template-partials) for code re-use. If this were **any** more complex, I would break it into a [component](https://ractive.js.org/concepts/#components) which would have it's own JS scope too.